azurerm/resource_arm_storage_management_policy_test.go

```diff
@@ -0,0 +1,207 @@
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.prefix_match.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.prefix_match.3439697764", "container1/prefix1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.blob_types.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.blob_types.1068358194", "blockBlob"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.tier_to_cool_after_days_since_modification_greater_than", "10"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.tier_to_archive_after_days_since_modification_greater_than", "50"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.delete_after_days_since_modification_greater_than", "100"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.snapshot.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.snapshot.0.delete_after_days_since_creation_greater_than", "30"),
+
+					// Rule2
+					resource.TestCheckResourceAttr(resourceName, "rule.1.name", "rule2"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.type", "Lifecycle"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.0.prefix_match.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.0.prefix_match.4102595489", "container2/prefix1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.0.prefix_match.1837232667", "container2/prefix2"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.0.blob_types.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.filters.0.blob_types.1068358194", "blockBlob"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.base_blob.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.base_blob.0.tier_to_cool_after_days_since_modification_greater_than", "11"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.base_blob.0.tier_to_archive_after_days_since_modification_greater_than", "51"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.base_blob.0.delete_after_days_since_modification_greater_than", "101"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.snapshot.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.1.actions.0.snapshot.0.delete_after_days_since_creation_greater_than", "31"),
+				),
+			},
+		},
+	})
+}
```

Comment:
Also we should have an Exists and CheckDestroy function to confirm that the resource exists and it's been destroyed.